tests/cypress/e2e/textFieldInitializerTest.cy.ts

```diff
@@ -0,0 +1,455 @@
+import {createSite, deleteSite} from '@jahia/cypress';
+import {PageComposer} from '../page-object/pageComposer';
+
+describe('Test the text field initializer)', () => {
+    const siteKey = 'extFieldInitializerTest';
+    const langEN = 'en';
+    const langFR = 'fr';
+    const langDE = 'de';
+    const languages = langEN + ',' + langFR + ',' + langDE;
+    const siteConfig = {
+        languages: languages,
+        templateSet: 'dx-base-demo-templates',
+        serverName: 'localhost',
+        locale: langEN
+    };
+
+    before(function () {
+        createSite(siteKey, siteConfig);
+        cy.runProvisioningScript({fileName: 'provisioning/enableModule.yaml', type: 'application/yaml'});
+    });
+
+    after(function () {
+        deleteSite(siteKey);
+    });
+
+    const checkFieldValues = (contentEditor, fields, lang) => {
+        // eslint-disable-next-line guard-for-in
+        for (const fieldId in fields) {
```

Comment:
This can be simplified to something like:
```

fields.forEach(field => {
            contentEditor.getField(field.type, field.key).addNewValue(field.values['en'])
        })
```

And I think you should update the array of field to something like
```

const fields = [{
                key: 'jnt:textFieldInitializer_defaultString',
                type: SmallTextField,
                values: {
                    en: 'Default string',
                    fr: 'Default string',
                    de: 'Default string'
                }
            },
            {
                key: 'jnt:textFieldInitializer_defaultI18nString',
                type: 'smallText',
                values: {
                    en: 'Default i18n string',
                    fr: 'Default i18n string',
                    de: 'Default i18n string'
                }
            }] ;


```

Reply:
And you should add a generic function in the Field component. 
  ```
  checkValue(string): void {
        // empty method
    }
```

and override it in each field object

Reply:
Fixed